apps/client/src/features/viewers/minimal-timer/MinimalTimer.tsx

```diff
@@ -133,14 +133,18 @@ export default function MinimalTimer(props: MinimalTimerProps) {
   const isNegative = (time.current ?? 0) < 0;
   const showFinished = isNegative && !userOptions?.hideOvertime;
   const showEndMessage = time.current < 0 && general.endMessage && !hideEndMessage;
+  const countUp =
+    time.duration - time.current > time.duration ? time.duration * -1 + time.current : time.duration - time.current;
+  const timer =
+    time.countdownStyle === 'count-down' ? time.current : time.countdownStyle === 'count-up' ? countUp : time.clock;
```

Comment:
it looks like we make this calculation in several places in the code.
The initial idea was to keep the code local, so people making their own integrations would have some examples without having to dig too far into the app.

I am not sure I agree with my old self on that anymore
I suggest we make a function **deriveTimerValues()** somewhere in a utils file (**src/common/utils/viewerUtils.ts** ?) which receives the timer object and returns an object with all these things we need.
We would apply it for now in **Timer.tsx** and **MinimalTimer.tsx**

This would also mean that we can easily make some tests on the logic here

Reply:
Sounds reasonable. I will leave the organisation to your judgement. 

However I would say that, if we deem that it has enough complexity, the logic that calculates this derived states should be extracted to a file and tested